control/timeresp.py

```diff
@@ -260,20 +260,29 @@ def forced_response(sys, T=None, U=0., X0=0., transpose=False,
     n_inputs = B.shape[1]
     n_outputs = C.shape[0]
 
+    # Convert inputs to numpy arrays for easier shape checking
+    if U is not None:
+        U = np.asarray(U)
+    if T is not None:
+        T = np.asarray(T)
+
     # Set and/or check time vector in discrete time case
     if isdtime(sys, strict=True):
         if T is None:
             if U is None:
                 raise ValueError('Parameters ``T`` and ``U`` can\'t both be'
                                  'zero for discrete-time simulation')
             # Set T to equally spaced samples with same length as U
-            T = np.array(range(len(U))) * (1 if sys.dt is True else sys.dt)
+            if len(U.shape) == 1:
+                n_steps = U.shape[0]
+            else:
+                n_steps = U.shape[1]
+            T = np.array(range(n_steps)) * (1 if sys.dt is True else sys.dt)
         else:
             # Make sure the input vector and time vector have same length
-            # TODO: allow interpolation of the input vector
```

Comment:
Leave this comment in, since at some point we would like to allow "subsampling"

Reply:
I have re-added this comment in the updated PR.